### PR TITLE
Disentangle Ethereum events logic from vote extensions

### PR DESCRIPTION
**Pull request by sug0**
Implements the first item of issue https://github.com/anoma/namada/issues/242

This process is quite boring, and I may not catch all references of vote extension which were really meant to reference Ethereum events. Therefore, future PRs may be needed to rename everything that's left.

### CODE REVIEW

**Review by james-chf — COMMENTED** [review]
We're adding a lot of TODO comments to the code, I think some of these could better come in (implemented) in the next PR.

File: apps/src/lib/node/ledger/shell/vote_extensions.rs [hunk 1]
```diff
@@ -198,22 +214,34 @@ mod extend_votes {
...
-        /// are invalid, in the form of a [`VoteExtensionError`].
+        /// are invalid, in the form of a [`EthEventsVextError`].
+        // TODO: the error type we return should be some kind of
+        // enum like:
+        //
+        // ```ignore
+        // enum VoteExtensionError {
+        //     EthEventsVext(EthEventsVextError),
+        //     ValidatorSetUpdateVext(ValidatorSetUpdateVextError),
+        // }
```
[the remark this answers is not in the document]
Comment (batconjurer) [reply apps/src/lib/node/ledger/shell/vote_extensions.rs:228]: I actually think that is a good point. At this point, I think using thiserror everywhere is just a matter of inertia. In some places it makes sense, but we ignore so many of the errors, or only log them.

File: apps/src/lib/node/ledger/shell/vote_extensions.rs [hunk 2]
```diff
@@ -2,34 +2,40 @@
...
 
     use super::super::queries::QueriesExt;
     use super::super::*;
 
-    /// A [`VoteExtension`] signed by a Namada validator.
-    pub type SignedExt = Signed<VoteExtension>;
-
     /// The error yielded from [`Shell::validate_vote_ext_and_get_it_back`].
     #[derive(Error, Debug)]
-    pub enum VoteExtensionError {
```
Comment (batconjurer) [inline apps/src/lib/node/ledger/shell/vote_extensions.rs:17]: I don't see a problem leaving this as an error type for all vexts.

**Review by batconjurer — APPROVED** [review]
Some small comments, but generally looks good

File: apps/src/lib/node/ledger/shell/vote_extensions.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (sug0) [reply apps/src/lib/node/ledger/shell/vote_extensions.rs:228]: yeah, I think I might delete this `EthEventsVextError` enum. the plan was to use it in `process_proposal`, but I'm just checking if things error out, I'm not actually returning the errors in a `TxResult`.

[the remark this answers is not in the document]
Comment (sug0) [reply apps/src/lib/node/ledger/shell/vote_extensions.rs:228]: nvm, I'm keeping `EthEventsVextError`, but renaming it back to `VoteExtensionError`. we can refactor some of these error enums we simply ignore in another PR. for now it's probably not worth the effort.

**Review by james-chf — APPROVED** [review]

**Pull request merged**